optuna/_hypervolume/utils.py

```diff
@@ -35,6 +37,7 @@ def _compute_2d(solution_set: np.ndarray, reference_point: np.ndarray) -> float:
     mask = sorted_solution_set[:, 1] <= reference_points_y_cummin
 
     used_solution = sorted_solution_set[mask]
-    edge_length_x = reference_point[0] - used_solution[:, 0]
-    edge_length_y = reference_points_y_cummin[mask] - used_solution[:, 1]
+    with np.errstate(invalid="ignore"):
+        edge_length_x = reference_point[0] - used_solution[:, 0]
+        edge_length_y = reference_points_y_cummin[mask] - used_solution[:, 1]
```

Comment:
Do we still need it after we added the guard? Shouldn't the warning happen at all because `inf - inf` will not happen?

Reply:
Sorry, you are right.